modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItIntervalTest.java

```diff
@@ -70,29 +70,29 @@ public void testIntervalResult() {
      */
     @Test
     public void testIntervalIntCast() {
-        assertThrows(IgniteException.class, () -> eval("CAST(NULL::INTERVAL SECONDS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(NULL::INTERVAL MONTHS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL 1 SECONDS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL 2 MINUTES AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL 3 HOURS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL 4 DAYS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL -4 DAYS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL 5 MONTHS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL 6 YEARS AS INT)"));
-        assertThrows(IgniteException.class, () -> eval("CAST(INTERVAL -6 YEARS AS INT)"));
+        assertNull(eval("CAST(NULL::INTERVAL SECONDS AS INT)"));
+        assertNull(eval("CAST(NULL::INTERVAL MONTHS AS INT)"));
+        assertEquals(1, eval("CAST(INTERVAL 1 SECONDS AS INT)"));
+        assertEquals(2, eval("CAST(INTERVAL 2 MINUTES AS INT)"));
+        assertEquals(3, eval("CAST(INTERVAL 3 HOURS AS INT)"));
+        assertEquals(4, eval("CAST(INTERVAL 4 DAYS AS INT)"));
+        assertEquals(-4, eval("CAST(INTERVAL -4 DAYS AS INT)"));
+        assertEquals(5, eval("CAST(INTERVAL 5 MONTHS AS INT)"));
+        assertEquals(6, eval("CAST(INTERVAL 6 YEARS AS INT)"));
+        assertEquals(-6, eval("CAST(INTERVAL -6 YEARS AS INT)"));
 
         assertEquals("+6", eval("CAST(INTERVAL 6 YEARS AS VARCHAR)"));
```

Comment:
Created an issue for casts from INTERVAL to VARCHAR because current results look a little bit strange - https://issues.apache.org/jira/browse/IGNITE-20238